cmd/openshift-install/create.go

```diff
@@ -290,53 +292,55 @@ func destroyBootstrap(ctx context.Context, config *rest.Config, directory string
 
 	eventTimeout := 30 * time.Minute
 	logrus.Infof("Waiting up to %v for the bootstrap-complete event...", eventTimeout)
-	eventContext, cancel := context.WithTimeout(ctx, eventTimeout)
+	if err := waitForEvent(ctx, events, "bootstrap-complete", eventTimeout); err != nil {
+		return err
+	}
+
+	logrus.Info("Destroying the bootstrap resources...")
+	return destroybootstrap.Destroy(rootOpts.dir)
+}
+
+// waitForEvent watches the events, waits for the event of the given name and prints out all other events on the way.
+func waitForEvent(ctx context.Context, client corev1client.EventInterface, name string, timeout time.Duration) error {
+	waitCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
-	_, err = Until(
-		eventContext,
-		"",
-		func(sinceResourceVersion string) (watch.Interface, error) {
-			for {
-				watcher, err := events.Watch(metav1.ListOptions{
-					ResourceVersion: sinceResourceVersion,
-				})
-				if err == nil {
-					return watcher, nil
-				}
-				select {
-				case <-eventContext.Done():
-					return watcher, err
-				default:
-					logrus.Warningf("Failed to connect events watcher: %s", err)
-					time.Sleep(2 * time.Second)
-				}
+
+	_, err := clientwatch.UntilWithSync(
+		waitCtx,
+		&cache.ListWatch{ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
```

Comment:
Is there a reason why we can't use `cache.NewListWatchFromClient` here instead of building our own `cache.ListWatch`?

Reply:
> Is there a reason why we can't use `cache.NewListWatchFromClient` here instead of building our own `cache.ListWatch`?

`NewListWatchFromClient` sounds good to me.  Want to work up a PR?

Reply:
Actually, I can work this up.